azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageRampupDaoImpl.java

```diff
@@ -0,0 +1,491 @@
+      + "irp.type_id = it.id and it.name = ? and irp.active = ? ";
+  private static String DEACTIVATE_ACTIVE_RAMPUP_PLAN_QUERY = "update image_rampup_plan "
+      + "set active = ? where id = ?";
+  private static String SELECT_IMAGE_RAMPUP_QUERY = "select ir.id, ir.plan_id, iv.version, "
+      + "ir.rampup_percentage, ir.stability_tag, ir.created_on, ir.created_by, ir.modified_on, "
+      + "ir.modified_by from image_versions iv, image_rampup_plan irp, image_rampup ir where "
+      + "irp.id = ir.plan_id and iv.id = ir.version_id and irp.id = ?";
+  private static String SELECT_ALL_IMAGE_TYPE_RAMPUP_QUERY = "select ir.id, ir.plan_id, "
+      + "iv.version image_version, ir.rampup_percentage, ir.stability_tag, it.name "
+      + "image_type_name, ir.created_on, ir.created_by, ir.modified_on, ir.modified_by from "
+      + "image_types it, image_versions iv, image_rampup_plan irp, image_rampup ir where "
+      + "irp.id = ir.plan_id and iv.id = ir.version_id and irp.type_id = it.id and irp.active = ? "
+      + "and iv.type_id = it.id and it.active = ?";
+
+
+  @Inject
+  public ImageRampupDaoImpl(DatabaseOperator databaseOperator, ImageTypeDao imageTypeDao,
+      final ImageVersionDao imageVersionDao) {
+    this.databaseOperator = databaseOperator;
+    this.imageTypeDao = imageTypeDao;
+    this.imageVersionDao = imageVersionDao;
+  }
+
+  @Override
+  public int createImageRampupPlan(ImageRampupPlanRequest imageRampupPlanRequest) {
+    ImageType imageType = imageTypeDao
+        .getImageTypeByName(imageRampupPlanRequest.getImageTypeName())
+        .orElseThrow(() -> new ImageMgmtDaoException("Unable to fetch image type metadata. Invalid "
+            + "image type : " + imageRampupPlanRequest.getImageTypeName()));
+    final SQLTransaction<Long> insertAndGetRampupPlanId = transOperator -> {
+      // Fetch the active rampup plan for the image type
+      Optional<ImageRampupPlan> optionalImageRampupPlan =
+          this.getActiveImageRampupPlan(imageType.getName());
+      /**
+       * If active rampup plan is already present and the activatePlan is set to true for the new
+       * plan, deactivate the current plan as only one rampup plan will be active at a time.
+       * Otherwise, appropriate error message will be thrown.
+       */
+      if (optionalImageRampupPlan.isPresent()) {
+        if (imageRampupPlanRequest.isActivatePlan()) {
+          transOperator.update(DEACTIVATE_ACTIVE_RAMPUP_PLAN_QUERY,
+              false, optionalImageRampupPlan.get().getId());
+        } else {
+          throw new ImageMgmtDaoException(String.format("Existing plan: %s is active for image "
+                  + "type: %s. To create and activate a new plan set activatePlan option to true.",
+              optionalImageRampupPlan.get().getPlanName(),
+              imageRampupPlanRequest.getImageTypeName()));
+        }
+      }
+      // Passing timestamp from the code base and can be formatted accordingly based on timezone
```

Comment:
We are keeping timestamp in all the new tables and same with experimentation API as well. This will keep all the APIs consistent.

Reply:
Agree. As far as timezone is used, it should be good. Otherwise based on data center timezone, it will have different timezone for each cluster in database.

Reply:
Yes all the Timezone specific change would be taken in the next PR. @ypadron-in  is planning to add one common class in the master branch.

Reply:
Oh nice. That will bring consistency. Can you please create a ticket to take care of integrating with that common class and list all the places where this integration will be needed?